src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/PodTemplateStepExecution.java

```diff
@@ -104,6 +113,50 @@ public void stop(Throwable cause) throws Exception {
         new PodTemplateAction(getContext().get(Run.class)).pop();
     }
 
+    private void collectAllowedClouds(Set<String> allowedClouds, ItemGroup<?> itemGroup) {
+    	if(itemGroup instanceof Folder) {
+    		Folder folder = (Folder) itemGroup;
+    		LOGGER.info("Inspecting folder: " + folder.getName());
+    		KubernetesFolderProperty p = folder.getProperties().get(KubernetesFolderProperty.class);
+    		
+    		String[] clouds = 
+    				StringUtils.split(p.getAllowedClouds(), "\n");
+    		if(clouds != null) {
+    			for(String cloud: clouds) {
+    				allowedClouds.add(cloud.trim());
+    			}
+    		}
+
+    		collectAllowedClouds(allowedClouds, folder.getParent());
+    	}
+    }
+    
+    private void checkAccess(Run<?, ?> run, KubernetesCloud kubernetesCloud) throws AbortException {
+        // Check if the usage of the KubernetesCloud has been restricted by requiring a secret.
+        if(kubernetesCloud.isUsageRestricted()) {
+        	Job<?, ?> job = run.getParent(); // Return the associated Job for this Build
+        	        	
+        	LOGGER.info("Job:" + job);
+        	LOGGER.info("JobStr: " + ReflectionToStringBuilder.toString(job));
+        	
+        	ItemGroup<?> parent = job.getParent(); // Inspect the Parent of the Job
+            
+    		Set<String> allowedClouds = new HashSet<>();
+            if(parent instanceof Folder) {
```

Comment:
Could recurse into folder hierarchy.

Reply:
```collectAllowedClouds``` is already traversing the hierarchy internally.